
# Individual dataset description

## Royer et al. 2021 E. coli BSI multicenter surveillance

### Opt for
- *Large scale epidemiology* session using already-assembled genomes
- Raw reads are not available

### Original WGS data description
- Target pathogen: E. coli from bloodstream infection cultures; all from Paris France; Collected in 2005 and 2016-2017.
- Sequencing platform: Illumina NextSeq
- N. isolates sequenced: 912 strains from 912 bloodstream infection patients 

### Dataset links
__NCBI BioProject__ accession numbers
- (1) 2005 isolates: PRJEB39260 --> [PRJEB39260 page](https://www.ncbi.nlm.nih.gov/bioproject/PRJEB39260/) 
- (2) 2016-2017 isolates: PRJEB35745 --> [PRJEB35745 page](https://www.ncbi.nlm.nih.gov/bioproject/PRJEB35745/) 

__Raw reads__ not available

__Genome assemblies__ assembly accession list txt
- (1) 2005 isolates: [NCBI assembly accession list file for 2005 isolates](https://github.com/kihyunee/genomics_tutorial/blob/main/datasets/Royer_2021/PRJEB39260_assembly_accession_list.txt)
- (2) 2016-2017 isolates: [NCBI assembly accession list file for 2016-2017 isolates](https://github.com/kihyunee/genomics_tutorial/blob/main/datasets/Royer_2021/PRJEB35745_assembly_accession_list.txt)

__Isolate metadata mapped__ to the genome assembly accessions
- tab-delimited text file containing the table: [isolate metadata table](https://github.com/kihyunee/genomics_tutorial/blob/main/datasets/Royer_2021/TableS2_isolate_metadata.txt)



# Selection of dataset per session

## In illumina-only assembly session


## In nanopore-only or hybrid assembly session


## In small dataset session


## In large dataset session



